ci/build_libtelio.py

```diff
@@ -317,7 +329,71 @@ def main() -> None:
         assert False, f"command '{args.command}' not supported"
 
 
+def try_download_artifacts(
+    tag_prefix, path_to_save, target_arch, target_os, moose=False
+):
+    def get_token():
+        if "LLH_GROUP_TOKEN_FLAKY_TESTS" in os.environ:
+            return os.environ["LLH_GROUP_TOKEN_FLAKY_TESTS"]
+        return getpass.getpass("Enter Gitlab API access token:")
+
+    token = get_token()
+
+    if target_os == "uniffi" and moose:
+        raise ValueError(
+            "Cannot download artifacts for uniffi and moose at the same time"
+        )
+
+    try:
+        if ArtifactsDownloader(
+            target_os,
+            target_arch,
+            token,
+            path_to_save,
+            tag_prefix,
+        ).download():
+            if moose and target_os in ["linux", "windows", "android"]:
+                moose_utils.fetch_moose_dependencies(target_os, MOOSE_MAP[target_arch])
+            return True
+        else:
+            print(f'Failed to download artifacts from "{tag_prefix}" pipeline')
+    except Exception as e:
+        print(f'Error while downloading artifacts from "{tag_prefix}" pipeline: {e}')
+
+    return False
+
+
+def exec_bindings(args):
+    if args.try_fetch_from_pipeline:
+        if try_download_artifacts(
+            args.try_fetch_from_pipeline,
+            PROJECT_ROOT,
+            target_arch=None,
+            target_os="uniffi",
+        ):
+            return
+
+    print("Proceeding with generating uniffi ...")
+
+    rutils.generate_uniffi_bindings(
+        PROJECT_CONFIG,
+        LIBTELIO_ENV_UNIFFI_GENERATORS_TAG,
+        ["python", "cs", "go", "swift", "kotlin"],
+        "src/libtelio.udl",
+        dockerized=args.dockerized,
+    )
+
+
 def exec_build(args):
+    # Try fetching from pipeline, if it fails, continue with the build
+    if args.try_fetch_from_pipeline:
+        if try_download_artifacts(
+            args.try_fetch_from_pipeline, PROJECT_ROOT, args.arch, args.os, args.moose
+        ):
+            return
+
+        print("Proceeding with build...")
```

Comment:
I think this is too passive and will be lost in between all the logs.

```suggestion
        while True:
        response = input("Artifacts download failed, do you want to proceed with the build? (y/n): ").strip().lower()
        if response == 'y':
            print("Proceeding with build...")
            break
        elif response == 'n':
            return;
        else:
            print("Invalid input. Please enter 'y' or 'n'.")
```

Reply:
Same as https://github.com/NordSecurity/libtelio/pull/543#discussion_r1616764373

Reply:
oh sry didn't see it.